File: js/easy/leetcode14.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
  if (strs.length == 0) {
    return "";
  }
  let ans = strs[0];
  let i = 1;
  while (i < strs.length) {
    if (!strs[i].startsWith(ans)) {
      ans = ans.slice(0, -1);
    } else {
      i++;
    }
  }
  return ans;
};
const strs = ["flower", "flow", "flight"];
longestCommonPrefix(strs);

File: c++/easy/leetcode14.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

string longestCommonPrefix(vector<string> &strs)
{
    if (strs.empty())
        return "" ;

    sort(strs.begin(), strs.end());

    string x = strs[0];
    string y = strs[strs.size() - 1];
    string ans = "";

    for (int i = 0; i < x.size(); i++)
    {
        if (x[i] == y[i])
            ans += x[i];
        else
            break;
    }
    return ans;
}

int main()
{
    vector<string> strs = {{""}};
    longestCommonPrefix(strs);
    return 0;
}